src/pages/Authentication.js

```diff
@@ -0,0 +1,48 @@
+import React, { memo, useState } from 'react';
+import { Redirect, useRouteMatch, Link } from 'react-router-dom';
+
+import { useDispatch, useSession, useStatus } from 'hooks';
+import AuthenticationForm from 'components/user/AuthenticationForm';
+import { login, register } from 'state/actions/userActions';
+import routes from 'constants/routesPaths';
+
+function AuthenticationPage() {
+  const { authenticated } = useSession();
+
+  const isLogin = !!useRouteMatch(routes.login);
+  const action = isLogin ? login : register;
```

Comment:
I know you want to reuse logic, with this Page, but I think it's better to split the context one for Login and another for Register that way you reduce the complexity of conditionals in the future and you won't face that many errors

Reply:
kk